infrastructure/src/main/java/org/corfudb/infrastructure/LogUnitServer.java

```diff
@@ -285,12 +284,35 @@ private void rangeWrite(CorfuPayloadMsg<RangeWriteMsg> msg,
         r.sendResponse(ctx, msg, CorfuMsgType.WRITE_OK.msg());
     }
 
+    /**
+     * Seal the server with the epoch by setting epochWaterMark.
+     *
+     * - The epochWaterMark is persisted to withstand restarts.
+     * - The epochWaterMark is inserted in the queue and then we wait to flush all operations
+     *   in the queue before this operation.
+     * - All operations after this operation but stamped with an older epoch will be failed.
+     */
+    @Override
+    public synchronized void sealServerWithEpoch(long epoch) {
+        if (epoch > serverContext.getLogUnitEpochWaterMark()) {
+            serverContext.setLogUnitEpochWaterMark(epoch);
+            batchWriter.waitForEpochWaterMark(epoch);
```

Comment:
Can you get rid of the getLogUnitEpochWaterMark? the server epoch is enough. This is bad code that we need to clean up. When the base server starts it can can seal on the other servers with the current server epoch.

Reply:
As discussed, will remove the getLogUnitEpochWaterMark from this method, but keep it in resetLogUnit to prevent multiple reset in same epoch.

Reply:
@zalokhan @Maithem  I removed epochWaterMark from sealing, removed flush logic from reset and also did some renaming. Please review my latest commit in this pr. I'll squash them later.